### PR TITLE
PMM-6797 CLI for HAProxy.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

**Comment by JiriCtvrtka** [comment]
This PR also contain small fix for https://jira.percona.com/browse/PMM-5888.
Because with empty metricsPath we got "/", not default one "/metrics".

PMM repo PR: https://github.com/percona/pmm/pull/680

**Review by askomorokhov — APPROVED** [review]

File: commands/inventory/add_service_haproxy.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	return &addHAProxyServiceResult{
+		Service: resp.Payload.Haproxy,
+	}, nil
+}
+
+// register command
+var (
+	AddHAProxyService  = new(addHAProxyServiceCommand)
+	AddHAProxyServiceC = addServiceC.Command("haproxy", "Add an haproxy service to inventory").Hide(hide)
```
Comment (percona-csalguero) [inline commands/inventory/add_service_haproxy.go]: to "the" inventory?

File: commands/inventory/add_service_haproxy.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+func init() {
+	AddHAProxyServiceC.Flag("name", "HAProxy service name. Required").Required().StringVar(&AddHAProxyService.ServiceName)
+	AddHAProxyServiceC.Flag("node-id", "HAProxy service node ID. Required").Required().StringVar(&AddHAProxyService.NodeID)
+	AddHAProxyServiceC.Flag("environment", "Environment name").StringVar(&AddHAProxyService.Environment)
+	AddHAProxyServiceC.Flag("cluster", "Cluster name").StringVar(&AddHAProxyService.Cluster)
+	AddHAProxyServiceC.Flag("replication-set", "Replication set name").StringVar(&AddHAProxyService.ReplicationSet)
+	AddHAProxyServiceC.Flag("custom-labels", "Custom user-assigned labels").StringVar(&AddHAProxyService.CustomLabels)
```
Comment (percona-csalguero) [inline commands/inventory/add_service_haproxy.go]: I don't want to be picky. it is that I just finished PMM-7473 and in that ticket I was asked to improve the help.
For example, in custom label I added `Example: region=east,app=app1` and for environment we could add an example like 'prod or qa'.
Also, we can use PlaceHolder to show the examples in the help.

File: commands/inventory/add_service_haproxy.go (hunk 1 above)
Comment (JiriCtvrtka) [reply commands/inventory/add_service_haproxy.go]: @percona-csalguero Fixed, thanks.

**Review by percona-csalguero — CHANGES_REQUESTED** [review]

File: commands/inventory/add_service_haproxy.go (hunk 2 above)
Comment (JiriCtvrtka) [reply commands/inventory/add_service_haproxy.go]: Yes, I can do same for HAProxy once will be your PR approved. To do that in same way.

Comment (JiriCtvrtka) [reply commands/inventory/add_service_haproxy.go]: @percona-csalguero Changed help in style of https://github.com/percona/pmm-admin/pull/136/files

**Review by percona-csalguero — APPROVED** [review]

File: commands/management/add_haproxy.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	AddHAProxyC.Flag("username", "HAProxy username").StringVar(&AddHAProxy.Username)
+	AddHAProxyC.Flag("password", "HAProxy password").StringVar(&AddHAProxy.Password)
+
+	AddHAProxyC.Flag("scheme", "Scheme to generate URI to exporter metrics endpoints").
+		PlaceHolder("http or https").StringVar(&AddHAProxy.Scheme)
+	AddHAProxyC.Flag("metrics-path", "Path under which metrics are exposed, used to generate URI").
+		PlaceHolder("/metrics").StringVar(&AddHAProxy.MetricsPath)
+	AddHAProxyC.Flag("listen-port", "Listen port of haproxy exposing the metrics for scraping metrics (Required)").Required().Uint16Var(&AddHAProxy.ListenPort)
+
+	AddHAProxyC.Flag("service-node-id", "Node ID where service runs (default is autodetected)").StringVar(&AddHAProxy.NodeID)
```
Comment (BupycHuk) [inline commands/management/add_haproxy.go]:
```suggestion
	AddHAProxyC.Flag("node-id", "Node ID (default is autodetected)").StringVar(&AddHAProxy.NodeID)
```

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**